File: MultiLanguageSandbox/src/test/deps/swift/48.swift
func powerOfTwo(_ n: Int) -> Int {
    return 1 << n
}


func testPowerOfTwo() {
    assert(powerOfTwo(3) == 8, "2^3 should be 8.")
    assert(powerOfTwo(5) == 32, "2^5 should be 32.")
    assert(powerOfTwo(10) == 1024, "2^10 should be 1024.")
    assert(powerOfTwo(0) == 1, "2^0 should be 1.")
    assert(powerOfTwo(15) == 32768, "2^15 should be 32768.")
    assert(powerOfTwo(30) == 1073741824, "2^30 should be 1073741824.")
}

// Call the test function to verify the functionality
testPowerOfTwo()

File: MultiLanguageSandbox/src/test/deps/elixir/48.exs
defmodule PowerOfTwo do
def power(n) when is_integer(n) and n >= 0 and n <= 30 do
    :math.pow(2, n) |> round()
  end
end



ExUnit.start()
defmodule PowerOfTwoTest do
  use ExUnit.Case

  test "power returns 1 for n = 0" do
    assert PowerOfTwo.power(0) == 1
  end

  test "power calculates 2^n for small n" do
    assert PowerOfTwo.power(1) == 2
    assert PowerOfTwo.power(2) == 4
    assert PowerOfTwo.power(3) == 8
  end

  test "power calculates 2^n for larger n" do
    assert PowerOfTwo.power(10) == 1024
    assert PowerOfTwo.power(20) == 1048576
  end

  test "power handles the maximum n value" do
    assert PowerOfTwo.power(30) == 1073741824
  end
end